File: backend/app/schemas/standing.py
from pydantic import BaseModel

from app.schemas.constructor import ConstructorBrief
from app.schemas.driver import DriverBrief


class DriverStandingOut(BaseModel):
    position: int
    driver: DriverBrief
    constructor_name: str
    points: float
    wins: int

    class Config:
        from_attributes = True


class ConstructorStandingOut(BaseModel):
    position: int
    constructor: ConstructorBrief
    points: float
    wins: int

    class Config:
        from_attributes = True


class PointsAfterRound(BaseModel):
    round: int
    points: float


class StandingsProgressionOut(BaseModel):
    driver: DriverBrief
    progression: list[PointsAfterRound]

    class Config:
        from_attributes = True

File: frontend/src/types/standing.ts
import type { DriverBrief, ConstructorBrief } from './index';

export interface DriverStandingOut {
  position: number;
  driver: DriverBrief;
  constructor_name: string;
  points: number;
  wins: number;
}

export interface ConstructorStandingOut {
  position: number;
  constructor: ConstructorBrief;
  points: number;
  wins: number;
}

export interface PointsAfterRound {
  round: number;
  points: number;
}

export interface StandingsProgressionOut {
  driver_ref: string;
  driver_name?: string;
  constructor_ref?: string;
  points_by_round: PointsAfterRound[];
}

// Aliases for components that expect different names
export type DriverStanding = DriverStandingOut;
export type ConstructorStanding = ConstructorStandingOut;
